app/src/tests/components/contents/manageAccess/ManageUsers.test.tsx

```diff
@@ -42,6 +42,8 @@ describe("ManageUsers", () => {
     await waitFor(() => {
       mockUsersWithPermissions.forEach((user) => {
         expect(screen.getAllByText(user.username, { exact: false })[0]).toBeVisible();
+        expect(screen.getAllByText(user.email ?? "no email", { exact: false })[0]).toBeVisible();
+        expect(screen.getAllByText(user.displayName ?? "no display name", { exact: false })[0]).toBeVisible();
```

Comment:
This isn't quite right I think, because if there are multiple users with no email, but only one of those users actually displays the correct text "no email", all of those users will still pass.

So can we do something like, first find the div containing the right username for the user (or the right id in the key?), then check that the correct text exists within that particular div?

Reply:
Also is the comment `    // only username roles are rendered` now outdated?